BDArmory/FX/ExplosionFX.cs

```diff
@@ -628,28 +644,45 @@ private void ExecuteBuildingBlastEvent(BuildingBlastHitEvent eventToExecute)
             //TODO: Review if the damage is sensible after so many changes
             //buildings
             DestructibleBuilding building = eventToExecute.Building;
-            building.damageDecay = 600f;
+            //building.damageDecay = 600f;
 
-            if (building)
+            if(building && building.IsIntact)
             {
                 var distanceFactor = Mathf.Clamp01((Range - eventToExecute.Distance) / Range);
-                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW * Power * distanceFactor;
-
-                damageToBuilding *= 2f;
+                float blastMod = 1;
+                switch (ExplosionSource)
+                {
+                    case ExplosionSourceType.Bullet:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW;
+                        break;
+                    case ExplosionSourceType.Rocket:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_ROCKET;
+                        break;
+                    case ExplosionSourceType.Missile:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_MISSILE;
+                        break;
+                    case ExplosionSourceType.BattleDamage:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BATTLE_DAMAGE;
+                        break;
+                }
+                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * blastMod * (Power * distanceFactor);
+                damageToBuilding /= 2;
```

Comment:
Might need some tweaking, damageToBuilding*2 was far too powerful, as was * 1, but going any lower and it will start taking considerable ordinance to bring down even the lighter buildings